lib/engine/part/revenue_center.rb

```diff
@@ -50,9 +50,25 @@ def max_revenue
       end
 
       def route_revenue(phase, train)
-        return @revenue[:diesel] if train.name.upcase == 'D' && @revenue[:diesel]
+        revenue_multiplier(train) * route_base_revenue(phase, train)
+      end
+
+      def route_base_revenue(phase, train)
+        return (@revenue[:diesel]) if train.name.upcase == 'D' && @revenue[:diesel]
+
+        phase.tiles.reverse.each { |color| return (@revenue[color]) if @revenue[color] }
+      end
+
+      def revenue_multiplier(train)
+        distance = train.distance
+        base_multiplier = train.multiplier.nil? ? 1 : train.multiplier
+
+        return base_multiplier if distance.is_a?(Numeric)
 
-        phase.tiles.reverse.each { |color| return @revenue[color] if @revenue[color] }
+        row = distance.index do |h|
+          h['nodes'].include?(type)
+        end
+        distance[row].fetch('multiplier', base_multiplier)
```

Comment:
why is there both a train multiplier and a distance multiplier?

Reply:
For a simple train that doubles everything:
```
{
  "name":"2D",
  "multiplier":2
  "price":200,
  "rusts_on":"4",
  "num":5
},
```

For a more complex implementation where only cities/offboards are doubled:
```
{
			"name": "4D",
			"distance": [
				{
					"nodes": [
						"city",
						"offboard"
					],
					"multiplier": 2,
					"pay": 4,
					"visit": 4
				},
				{
					"nodes": [
						"town"
					],
					"pay": 99,
					"visit": 99
				}
			],
			"available_on": "5",
			"price": 650,
			"num": 3
		},
```

The final line ensures that the train multiplier is respected if the multiplier is not explicitly included in the distance section.